docs/designs/table-of-contents.md

```diff
@@ -0,0 +1,149 @@
+## [Header1.1](href)
+# Header2
+## [Header2.1](href)
+### [Header2.1.1](href)
+## [Header2.2](href)
+# @UidForAnArticle
+```
+
+For a *TOC Item*, it can be either plain text, or a [Markdown inline link](http://daringfireball.net/projects/markdown/syntax#link), or `@uid` as the shortcut for cross reference.
+
+Three kinds of links are supported:
+
+1. Absolute path, for example, `http://example.net`.
+2. Relative path, for example, `../example.md`. This kind of link has several advanced usages and is described in detail [below](#toc-item-link-href).
+3. URI with `xref` scheme, for example, `xref:System.String`, the value is the `uid` of the file to be cross referenced.
+
+## YAML format TOC `toc.yml`
+
+It represents a structured data model and conforms to the [YAML standard](http://www.yaml.org/spec/1.2/spec.html)
+
+```yml
+- name: Topic1
+  href: Topic1.md
+- name: Topic2
+  href: Topic2.md
+  items:
+    - name: Topic2_1
+      href: Topic2_1.md
+```
+
+### Data model for `toc.yml`
+
+The data model for `toc.yml` is an **array** of *TOC Item Object*.
+
+### *TOC Item Object*
+
+*TOC Item Object* represents the data model for each *TOC Item*.
+
+> [!Note]
+> All the property names are **case sensitive**.
+
+#### Properties
+
+Property Name                    | Type                       | Description
+-------------------------------- | -------------------------- | ---------------------------
+*name*                           | string                     | Specifies the title of the *TOC Item*.
+*href*                           | string                     | Specifies the hyperlink of the *TOC Item*. When its value is another TOC file, it is a TOC include, and all the items inside the included toc are considered as the child of current *TOC Item*
+*items*                          | array of *TOC Item Object* | Specifies the children *TOC Items* of current *TOC Item*.
+*topicHref*<sub>*advanced*</sub> | string                     | Specifies the topic href of the *TOC Item*. It is useful when *href* is linking to a folder or *tocHref* is used.
+*topicUid*<sub>*advanced*</sub>  | string                     | Specifies the `uid` of the *topicHref* file. If the value is set, it overwrites the value of *topicHref*.
+
+> [!Note]
+> *advanced*: These properties are useful when a TOC links another TOC, or links to a uid.
+
+## Toc Item Link `href`
+
+If a *TOC Item* is linking to some relative path, there are three cases:
+
+1. Linking to another *TOC File*, for example, `href: examples/toc.md`.
+2. Linking to a folder, which means, the value of the link ends with `/` explicitly, for example, `href: examples/`
+3. Linking to some local file.
+
+Each case is described in detail below.
+
+### Link to another *TOC File*
+
+If the *TOC Item* is linking to some other *TOC File*, it is considered as a placeholder of the referenced *TOC File*, and DocFX will extract content from that *TOC File* and insert into current *TOC Item* **recursively**.
```

Comment:
It is not a simple *file content embed*, href in included toc should be resolved relative to that included toc if there are in two different folder.

Reply:
you are right, it will be resolved to root toc, but is that need to be exposed to the end user? i believe it's a implementation logic

---
In reply to: [184844899](https://github.com/dotnet/docfx/pull/2703#discussion_r184844899) [](ancestors = 184844899)